### DISCUSSION TITLE
Duplicate posting when replying with media

### DISCUSSION THREAD

**Discussion by michaellwatson** [danieldevine/bird-elephant#40] [Q&A] [answered]
Hi there, great library and perfect for what I need, although I wondered if you could help me with one issue:

I want to reply to a tweet/mention with media, it works but its posting twice using the code below, is this supported or am I doing something wrong? please let me know, thanks in advance

`$image = $twitter->tweets()->upload('./skypicture.jpg');

    // //pass the returned media id to a media object as an array
    $media = (new \Coderjerk\BirdElephant\Compose\Media)->mediaIds(
        [
            $image->media_id_string
        ]
    );

    $reply = (new \Coderjerk\BirdElephant\Compose\Reply)->inReplyToTweetId('1551627359270457344');

    $tweet = (new \Coderjerk\BirdElephant\Compose\Tweet)->media($media)->text('Test Tweet Reply')->reply($reply);

    $twitter->tweets()->tweet($tweet);`

**Comment by danieldevine** [comment, marked as answer]
Hi @michaellwatson , 

I just tried it out there and it seems to have worked as expected: https://twitter.com/AliFerguson2022/status/1551627359270457344 

Maybe there's some other issue - is there more code you can show?

**Reply by michaellwatson** [reply]
@danieldevine big thanks for your help, turns out it was an issue with chrome, switched firefox and it worked great :)

**Reply by danieldevine** [reply]
Excellent, glad it worked out.

**Comment by joneslisboa** [comment]
Hello, I have the same problem, when posting with media, the code I use, and the answer below, I would like to understand if it is my error or something in the api

  $media = $param['media'];

         $twitter = new BirdElephant($credentials);
         $upload = $twitter->tweets()->upload($media);
         $media = (new \Coderjerk\BirdElephant\Compose\Media)->mediaIds(
             [
                 $upload->media_id_string
             ]
         );

         $tweet = (new \Coderjerk\BirdElephant\Compose\Tweet)->text($param['texto'])
             ->media($media);
         $twitter->tweets()->tweet($tweet);
         $result = $twitter->tweets()->tweet($tweet);
      
         return $result;


Client error: `POST https://api.twitter.com/2/tweets` resulted in a `403 Forbidden` response:
{"detail":"You are not allowed to create a Tweet with duplicate content.","type":"about:blank","title":"Forbidden","stat (truncated...)

**Reply by joneslisboa** [reply]
however the tweet and media are posted...

**Reply by michaellwatson** [reply]
@joneslisboa it looks like your calling the function x2, could remove this line $twitter->tweets()->tweet($tweet);

**Reply by joneslisboa** [reply]
Perfect my friend, thank you very much, I was the one who was stupid... :)

**Reply by michaellwatson** [reply]
@joneslisboa welcome :) don't say that, its easy to overlook stuff, especially if you been looking at the code for a while
